apps/studio/src/components/PageEditor/ComponentSelector.tsx

```diff
@@ -117,10 +128,15 @@ function ComponentSelector() {
     const nextPageState = !!newComponent
       ? [...savedPageState, newComponent]
       : savedPageState
-    setSavedPageState(nextPageState)
+
     setDrawerState({ state: nextState })
     setCurrActiveIdx(nextPageState.length - 1)
     setPreviewPageState(nextPageState)
+
+    // TODO: Decide if this is a good idea...
```

Comment:
what's this todo? what are the alternatives to evaluate if this is a good idea?

Reply:
i'll add more context to the comment (and here): right now, we're doing an `if` condition on `setAddedBlock` because we want the typing to be correct (ie, `addedBlock` is only for complex blcoks, which disallows `prose`) 

what this means practically is that we get alot of `if/else` scattered over the place. if we combine, we'd need to handle the `prose` case for components. 

i decided on this approach cos it's simple for now